app/Services/PostService.php

```diff
@@ -105,6 +104,13 @@ public function destroy($postId)
             'id' => $postId,
         ]);
 
-        return $this->repository->destroy($postId);
+        $trashed = $this->repository->destroy($postId);
+
+        // Dispatch job to send post to Quasar
+        if (config('features.pushToQuasar')) {
+            SendPostToQuasar::dispatch(Post::withTrashed()->find($postId));
```

Comment:
Is this how we want to handle situations like this, and then Quasar differentiates based on `deleted_at`? I'm tempted to think it's a bit cleaner to send them a separate delete event saying "we deleted a X with this ID".

cc: @sheyd

Reply:
JK JK, see the existing discussion on this [above](https://github.com/DoSomething/rogue/pull/603/files/c4611d85313825e08f8ba6964de99f4692233c3c..d42b1d2a7db33f9f2b9ada9e01d46e8c36760288#r165102243).

Reply:
@DFurnes This is what I talked to Sena about but I also didn't think of that! @sheyd if Dave's suggestion sounds better then we can certainly do that!

Reply:
@DFurnes I've been coding the consumer on the assumption that if a `deleted_at` date is present, we'll handle the record updates on the Quasar side. I'm happy to accommodate whatever changes you think make sense here, but at the moment, I'm fine with sticking with what we have here. 

On the Quasar side, what I'm checking on per event is if `deleted_at` is NULL. If not, then it triggers a record delete function in the Quasar data warehouse (we don't actually delete the record, just mark status as `deleted`). This entails a lookup per Rogue event for an actual value, which I'm doing for the Rogue queue routing code. Key lookups generally are more efficient, so if something like `[data][delete_post]` exists vs checking for value, I'm happy to go with that, but I'm happy to live with the current performance hit of per event value check and throw more hardware at this, since we're using so little CPU in general, if it saves Rogue code increasing in complexity.

Reply:
Just talked this over with Sena IRL – let's keep things in Quasar & Rogue working exactly as we've got 'em, but just send this payload for a "deleted" event (rather than re-loading the full soft-deleted record from the database):

```php
'data' => [
  'id' => $postId,
  'deleted_at' => Carbon::now(),
],
'meta' => [
  'message_source' => 'rogue',
]
```

We can always re-visit if there's a better general "contract" for this for a v1.1.

Reply:
@DFurnes Okay! It's okay that the `deleted_at` timestamp we send here may not match the `deleted_at` on the post in Rogue?

Reply:
I've updated Rogue consumer code on Quasar side to account for these changes.

Reply:
> Okay! It's okay that the `deleted_at` timestamp we send here may not match the `deleted_at` on the post in Rogue?

@katiecrane Yeah, I think so. It should be pretty darn close.